composables/useFees.ts

```diff
@@ -0,0 +1,54 @@
+import type { Address } from 'viem'
+import type { Prefix } from '@kodadot1/static'
+import type { Actions } from '@/composables/transaction/types'
+
+type GetFeeParams = {
+  prefix: Prefix
+  action: Actions
+}
+
+export default function () {
+  const { accountId } = useAuth()
+
+  const getSubstrateTransactionFee = async ({ action }: GetFeeParams): Promise<number> => {
+    const { apiInstance } = useApi()
+    const api = await apiInstance.value
+    const fee = await getActionTransactionFee({
+      action: { ...action, address: accountId.value } as Actions,
+      address: accountId.value,
+      api,
+    })
+    return Number(fee)
+  }
+
+  // TODO support getActionTransactionFee on evm
+  const getEvmTransactionFee = async ({ prefix }: GetFeeParams): Promise<number> => {
+    const { publicClient } = useViem(prefix)
+
+    const estimatedGas = await publicClient.estimateGas({
+      to: accountId.value as Address,
+      value: BigInt('1000000000000000000'),
+      data: '0x',
+    })
```

Comment:
Is the estimated gas close to the real one?

Reply:
added support for evm on `getActionTransactionFee `, now the transaction fee is exact